editor/editor_data.cpp

```diff
@@ -534,6 +534,7 @@ Variant EditorData::instantiate_custom_type(const String &p_type, const String &
 				if (n) {
 					n->set_name(p_type);
 				}
+				n->set_meta("__custom_type_script", script);
```

Comment:
Meta keys use StringName, so `"__custom_type_script"` should either use SNAME, or better, be registered in core string names.